client/layout/index.jsx

```diff
@@ -71,6 +71,22 @@ const Layout = createReactClass( {
 		colorSchemePreference: PropTypes.string,
 	},
 
+	getInitialState: function() {
+		return { translator };
+	},
```

Comment:
Hi @ramonjd , thank you for your review! I thought of this solution before, but my concern is that we can't guarantee the execution order of event handlers. `communityTranslatorJumpstart.init` must be executed before TranslatorLauncher's `forceUpdate`, but in my understanding, there's no way for us specify that order. If `forceUpdate` is called before `communityTranslatorJumpstart.init`, TranslatorLauncher will still render with the old translator data. 😟

Reply:
Sorry for the late!
First of all, I think passing `translator.isEnabled()` and `translator.isActivated()` as props to `<TranslatorLauncher />` is fundamentally wrong, since

1. These two functions relying on the underlying pub/sub model to be correct. 
2. [<TranslatorLauncher /> has imported](https://github.com/Automattic/wp-calypso/blob/9bf86c5b067ac98569d8fe85a5a99e05c326a2d6/client/layout/community-translator/launcher.jsx#L15) it and used it already.

I'm agreeing with @ramonjd here to not introduce any more pub/sub to the code, so the best we could do is to reuse what we have and encapsulate them as much as possible. 

One possible way to approach this is to:

1. Pass in the whole `translator` as a prop or just use the imported one, and update the prop references accordingly.
1. Register a redraw function on `userSettings.on( 'change' )`. I personally think this makes more sense than `i18n.on( 'change' )` since showing `<TranslatorLauncher />` or not depends on the user settings, not necessarily the i18n state changes. It's possible that the redrawing function got executed before `communityTranslatorJumpstart.init()`, but I think that should be okay since the jumpstart object is not caching anything from user settings. Thus, the previous step should be able to make sure we are accessing the fresh data.

I haven't really tested it, and there could be numbers of better ways approaching this. Either way, I think the baseline is to not introduce new event emitters and to encapsulate the logic into `<TranslatorLauncher />`.

Does that make sense to y'all?